### PR TITLE
Use Prettier and our plugin to format files

### PR DESCRIPTION
**Pull request by Princesseuh**
## Changes

Move from using VS Code's formatters to Prettier and our plugin, which going forward will be our official way to format Astro files. We bundle Prettier and the plugin in the language server unless the user has those installed in which case we load theirs

This should solve all the issues related to formatting people had, further issues related to formatting will be contained in https://github.com/withastro/prettier-plugin-astro/issues

## Testing

Modified tests for this. This is slightly inconvenient to test for in a monorepo like this, so we'll mostly test the basic things. Testing behaviour is left to the plugin itself

## Docs

Updated commands and README

### CODE REVIEW

File: .changeset/tidy-adults-arrive.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@astrojs/language-server': minor
```
Comment (matthewp) [inline .changeset/tidy-adults-arrive.md:2]: should it be major?

Comment (Princesseuh) [reply .changeset/tidy-adults-arrive.md:2]: Language server is pre 1.0 still!

File: packages/language-server/src/importPackage.ts [hunk 2]
```diff
@@ -48,3 +49,14 @@ export function importSvelteIntegration(fromPath: string): typeof svelte | undef
...
 }
+
+export function importPrettier(fromPath: string): typeof prettier {
+	// This shouldn't ever fail, because we bundle Prettier in the extension itself
+	const prettierPkg = getPackagePath('prettier', [fromPath, __dirname])!;
+
+	return require(prettierPkg);
+}
+
+export function importPrettierPlugin(fromPath: string): string {
```
Comment (matthewp) [inline packages/language-server/src/importPackage.ts:60]: More of a function naming thing, but I find it a little unexpected that importPrettier imports it into the module graph but this one does not.

File: .changeset/tidy-adults-arrive.md (hunk 1 above)
Comment (matthewp) [reply .changeset/tidy-adults-arrive.md:2]: Oh, right, forgot about that.

File: packages/language-server/src/importPackage.ts (hunk 2 above)
Comment (matthewp) [reply packages/language-server/src/importPackage.ts:60]: This isn't blocking to me, just an observation.

**Review by matthewp — APPROVED** [review]

**Pull request merged**